tests/parametric/test_span_links.py

```diff
@@ -161,33 +184,36 @@ def test_span_link_from_distributed_w3c_headers(self, test_agent, test_library):
         assert link.get("flags") == 1 | -2147483648
         assert len(link.get("attributes") or {}) == 0
 
-    @missing_feature(library="python", reason="test not implemented")
+    #
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_API_VERSION": "v0.5"}, {"DD_TRACE_API_VERSION": "v0.4"}])
     def test_span_with_attached_links(self, test_agent, test_library):
         """Test adding a span link from a span to another span.
         """
         with test_library:
             with test_library.start_span("root") as parent:
                 with test_library.start_span("first", parent_id=parent.span_id) as first:
                     pass
-                with test_library.start_span(
-                    "second", parent_id=parent.span_id, links=[Link(parent_id=parent.span_id)]
-                ) as second:
-                    second.add_link(first.span_id, attributes={"bools": [True, False], "nested": [1, 2]})
 
-        traces = test_agent.wait_for_num_traces(1)
-        trace = find_trace_by_root(traces, Span(name="root"))
-        assert len(trace) == 3
+            with test_library.start_span("second") as second:
+                second.add_link(trace_id=parent.trace_id, span_id=parent.span_id)
+                second.add_link(
+                    trace_id=first.trace_id,
+                    span_id=first.span_id,
+                    attributes={"bools": [True, False], "nested": [1, 2]},
+                )
```

Comment:
add_link intentionally has no trace_id. Users of the API are already expected by other functions to have a mapping <span-id, span> in their test suite. The test is meant to create a link from a given span to another. This may be achieved by a dedicated API, by creating a context from the span, by whatever the preferred method in your language is, but the test itself doesn't directly require a capability "create span link with this span id and trace id", but "create span link to this span".

Reply:
Yes this is ideal but we could run into this scenario: 
Span A -> is created, finished, and flushed
Span B -> is created and attempts to link to Span A
  - Span B can not reference span A in a link because it it no longer stored in the client app.

The only way to guarantee this does not happen is to ensure Span A and Span B are both active. In some client apps this means that Span A will always be the parent of Span B. 

In my refactor I was trying to create a scenarios where Span A and Span B can be started/finished independently and the information about the trace_id is not lost. 

Is this a valid concern? If this is not a valid concern, then we can just close this PR

Reply:
No it isn't, it's just relevant for the purposes of testing - in a real application, the user would actually hold a reference to a span or spancontext keeping the span(context) in memory even after it's flushed.